### ISSUE TITLE
[i18n] Unexpected behaviors

### ISSUE THREAD

**Issue by rafaelss95** [angular-eslint/angular-eslint#513]
**Description and reproduction of the issue**

1. `boundTextAllowedPattern` isn't behaving as expected. See this case:
https://github.com/angular-eslint/angular-eslint/blob/b434b513abee0b3bee4db64c8fecfd866324634b/packages/eslint-plugin-template/tests/rules/i18n.test.ts#L204-L215

Here we report a bound text that starts with an allowed character. It should be valid instead;

2. Missing i18n-* attributes are reported duplicately. See this case:

https://github.com/angular-eslint/angular-eslint/blob/b434b513abee0b3bee4db64c8fecfd866324634b/packages/eslint-plugin-template/tests/rules/i18n.test.ts#L167-L178

3. Report ranges could be more assertive. For example, when an i18n attribute is missing an id or the text is missing the i18n attribute, the whole element is reported, when we just need to report the original attribute or text;

4. `<ng-template>` , `ngPlural`, `ngPluralCase` and `ngProjectAs` should be ignored;

5. Numeric values and HTML entities (https://github.com/mgechev/codelyzer/issues/618 and https://github.com/mgechev/codelyzer/issues/952) should be ignored. Examples:

```html
<div ariaselected="0"></div>
<div matBadge="&#8288;">5</div>
<div>732</div>
<div>&nbsp;</div>
```

6. ICUs are not being reported (https://angular.io/guide/i18n#translate-plurals-and-alternate-expressions);

7. Duplicate custom IDs are not being reported (https://angular.io/guide/i18n#define-unique-custom-ids).
